src/otx/core/data/dataset/base.py

```diff
@@ -257,7 +263,97 @@ def _cache_img(self, key: str | int, img_data: np.ndarray, meta: dict[str, Any]
     def _get_item_impl(self, idx: int) -> TorchDataItem | None:
         pass
 
-    @property
-    def collate_fn(self) -> Callable:
-        """Collection function to collect KeypointDetDataEntity into KeypointDetBatchDataEntity in data loader."""
-        return TorchDataItem.collate_fn
+    def get_collate_fn(self) -> Callable:
+        """Get collate function based on mode and stack_images flag."""
+        if self.collate_mode == "torch":
+            return TorchDataItem.collate_fn
+        if self.collate_mode == "numpy":
+            return NumpyDataItem.collate_fn
+
+        msg = f"Invalid collate mode: {self.collate_mode}"
+        raise ValueError(msg)
+
+    def postprocess(
+        self,
+        image: np.ndarray,
+        img_info: ImageInfo,
+        label: np.ndarray | None = None,
+        bboxes: np.ndarray | None = None,
+        masks: np.ndarray | None = None,
+        polygons: list[dict[str, Any]] | None = None,
+        keypoints: np.ndarray | None = None,
+    ) -> T_OTXDataEntity:
+        """Postprocess the data item.
+
+        This function converts the data item to the appropriate format and applies
+        the necessary transformations if needed.
+
+        Args:
+            image: Image data.
+            label: Label data.
+            masks: Mask data.
+            bboxes: Bounding box data.
+            keypoints: Keypoint data.
+            polygons: Polygon data.
+            img_info: Image information.
+
+        Returns:
+            A data item with the provided data.
```

Comment:
Maybe we should mention here that `postprocess` does not apply transforms when using numpy data item.